src/software/amazon/ion/Timestamp.java

```diff
@@ -1854,18 +1944,37 @@ public void print(Appendable out)
         print(out, adjusted);
     }
 
+    /**
+     * Safe version of print failing if the fractional second scale is bigger than a safe threshold to
+     * avoid hanging.
+     *
+     * @throws UnsupportedOperationException if the fraction second scale is bigger than a safe threshold.
+     *
+     * @see #print(Appendable)
+     */
+    public void safePrint(Appendable out)
+        throws IOException
+    {
+        checkFractionScale();
+        print(out);
+    }
 
     /**
      * Prints to an {@code Appendable} the string representation (in Ion format)
      * of this Timestamp in UTC.
      * <p>
      * This method produces the same output as {@link #toZString()}.
      *
+     * <br>
+     * <strong>WARNING:</strong> this method can hang for Timestamps with a very large
```

Comment:
I think this is warning too dire; any number with extremely high precision can "hang".   A better place for a warning is on the Timestamp factory method, where we can point out the impact of large negative exponents.

Also, "hang" is not a well-defined term so it should be avoided.